hpolib/benchmarks/ml/svm_benchmark.py

```diff
@@ -0,0 +1,316 @@
+import time
+from typing import Union, Tuple, Dict, List
+
+import numpy as np
+import ConfigSpace as CS
+
+from scipy import sparse
+from sklearn import svm
+from sklearn import pipeline
+from sklearn.compose import ColumnTransformer
+from sklearn.impute import SimpleImputer
+from sklearn.metrics import accuracy_score, make_scorer
+from sklearn.preprocessing import OneHotEncoder
+
+import hpolib.util.rng_helper as rng_helper
+from hpolib.abstract_benchmark import AbstractBenchmark
+from hpolib.util.openml_data_manager import OpenMLHoldoutDataManager
+
+__version__ = '0.0.1'
+
+
+class SupportVectorMachine(AbstractBenchmark):
+    """
+    Hyperparameter optimization task to optimize the regularization
+    parameter C and the kernel parameter gamma of a support vector machine.
+    Both hyperparameters are optimized on a log scale in [-10, 10].
+    The X_test data set is only used for a final offline evaluation of
+    a configuration. For that the validation and training data is
+    concatenated to form the whole training data set.
+    """
+
+    def __init__(self, task_id: Union[int, None] = None,
+                 rng: Union[np.random.RandomState, int, None] = None):
+        """
+        Parameters
+        ----------
+        task_id : int, None
+        rng : np.random.RandomState, int, None
+        """
+        super(SupportVectorMachine, self).__init__(rng=rng)
+
+        self.task_id = task_id
+        self.accuracy_scorer = make_scorer(accuracy_score)
+
+        self.X_train, self.y_train, self.X_valid, self.y_valid, self.X_test, self.y_test, variable_types = \
+            self.get_data()
+        self.categorical_data = np.array([var_type == 'categorical' for var_type in variable_types])
+
+        # Sort data (Categorical + numerical) so that categorical and continous are not mixed.
+        categorical_idx = np.argwhere(self.categorical_data)
+        continuous_idx = np.argwhere(~self.categorical_data)
+        sorting = np.concatenate([categorical_idx, continuous_idx]).squeeze()
+        self.categorical_data = self.categorical_data[sorting]
+        self.X_train = self.X_train[:, sorting]
+        self.X_valid = self.X_valid[:, sorting]
+        self.X_test = self.X_test[:, sorting]
+
+        nan_columns = np.all(np.isnan(self.X_train), axis=0)
+        self.categorical_data = self.categorical_data[~nan_columns]
+
+        mean_imputer = SimpleImputer(strategy='mean')
```

Comment:
This should be part of the pipeline as well and only done on the given fraction of the training data